ee/src/main/java/org/jboss/as/ee/structure/InitializeInOrderProcessor.java

```diff
@@ -49,24 +50,40 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         if (parent == null) {
             return;
         }
+
         final EarMetaData earConfig = parent.getAttachment(Attachments.EAR_METADATA);
-        if (earConfig != null) {
-            final boolean inOrder = earConfig.getInitializeInOrder();
-            if (inOrder && earConfig.getModules().size() > 1) {
+        final List<String> startupBeanModules = deploymentUnit.getParent().getAttachmentList(org.jboss.as.ee.component.Attachments.DEPLOYMENT_STARTUP_MODULES);
```

Comment:
`parent` is already available above, so no need for `deploymentUnit.getParent()`.